heat/core/tensor.py

```diff
@@ -271,12 +293,84 @@ def max(self, axis=None, out=None):
 
         return operations.max(self, axis, out)
 
-    def mean(self, axis):
-        # TODO: document me
-        # TODO: test me
-        # TODO: sanitize input
-        # TODO: make me more numpy API complete
-        return self.sum(axis) / self.shape[axis]
+    def mean(self, dimen=None, all_procs=False):
+        """
+        Calculates and returns the mean of a tensor.
+        If a dimension is given, the mean will be taken in that direction.
+
+        Parameters
+        ----------
+        self : ht.tensor
+            Values for which the mean is calculated for
+        dimen : None, Int, iterable
+                Dimension which the mean is taken in.
+                Default: None -> mean of all data calculated
+        all_procs : Bool
+                    Flag to distribute the data to all processes
+                    If True: will split the result in the same direction as x
+                    Default: False (mean of the whole dataset still calculated but not on every node)
+
+        Returns
+        -------
+        ht.tensor containing the mean/s, if split, then split in the same direction as x.
+        """
+        return operations.mean(self, dimen, all_procs)
+
+    def var(self, dimen=None, all_procs=False, bessel=True):
```

Comment:
Should expose the numpy.var() interface (https://docs.scipy.org/doc/numpy/reference/generated/numpy.var.html)